FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1494 @@
+ *
+ * @param[in] pxContext Context of the initial command.
+ */
+static void prvCommandCallback( CommandContext_t * pxContext );
+
+/**
+ * @brief The task used to create various publish operations.
+ *
+ * @param[in] pvParameters Parameters as passed at the time of task creation. Not
+ * used in this example.
+ */
+void prvPublishTask( void * pvParameters );
+
+/**
+ * @brief The task used to wait for incoming publishes.
+ *
+ * @param[in] pvParameters Parameters as passed at the time of task creation. Not
+ * used in this example.
+ */
+void prvSubscribeTask( void * pvParameters );
+
+/**
+ * @brief The main task used in the MQTT demo.
+ *
+ * @param[in] pvParameters Parameters as passed at the time of task creation. Not
+ * used in this example.
+ */
+static void prvMQTTDemoTask( void * pvParameters );
+
+/**
+ * @brief The timer query function provided to the MQTT context.
+ *
+ * @return Time in milliseconds.
+ */
+static uint32_t prvGetTimeMs( void );
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Global MQTT context.
+ */
+static MQTTContext_t globalMqttContext;
+
+/**
+ * @brief List of operations that are awaiting an ack from the broker.
+ */
+static AckInfo_t pxPendingAcks[ PENDING_ACKS_MAX_SIZE ];
+
+/**
+ * @brief List of active subscriptions.
+ */
+static SubscriptionElement_t pxSubscriptions[ SUBSCRIPTIONS_MAX_COUNT ];
+
+/**
+ * @brief Queue for main task to handle MQTT operations.
+ */
+static QueueHandle_t xCommandQueue;
+
+/**
+ * @brief Response queue for prvPublishTask.
+ */
+static QueueHandle_t xPublisherResponseQueue;
+
+/**
+ * @brief Response queue for prvSubscribeTask.
+ */
+static QueueHandle_t xSubscriberResponseQueue;
+
+/**
+ * @brief Response queue for publishes received on non-subscribed topics.
+ */
+static QueueHandle_t xDefaultResponseQueue;
+
+/**
+ * @brief Handle for prvMQTTDemoTask.
+ */
+static TaskHandle_t xMainTask;
+
+/**
+ * @brief Handle for prvPublishTask.
+ */
+static TaskHandle_t xPublisherTask;
+
+/**
+ * @brief Handle for prvSubscribeTask.
+ */
+static TaskHandle_t xSubscribeTask;
+
+/**
+ * @brief The network buffer must remain valid for the lifetime of the MQTT context.
+ */
+static uint8_t buffer[ NETWORK_BUFFER_SIZE ];
+
+/**
+ * @brief Global entry time into the application to use as a reference timestamp
+ * in the #prvGetTimeMs function. #prvGetTimeMs will always return the difference
+ * between the current time and the global entry time. This will reduce the chances
+ * of overflow for the 32 bit unsigned integer used for holding the timestamp.
+ */
+static uint32_t ulGlobalEntryTimeMs;
+
+/*-----------------------------------------------------------*/
+
+/*
+ * @brief Create the task that demonstrates the MQTT Connection sharing demo.
+ */
+void vStartSimpleMQTTDemo( void )
+{
+    /* This example uses a single application task, which in turn is used to
+     * connect, subscribe, publish, unsubscribe and disconnect from the MQTT
+     * broker. */
```

Comment:
This description about single task may be wrong.

Reply:
Changed